src/Artsy/Router/NetworkOfflineMonitor.tsx

```diff
@@ -0,0 +1,38 @@
+import { ModalDialog } from "Components/Modal/ModalDialog"
+import React, { useEffect, useState } from "react"
+
+export const NetworkOfflineMonitor: React.FC = () => {
+  const [showOfflineModal, toggleOfflineModal] = useState(false)
+
+  const setOffline = () => {
+    toggleOfflineModal(true)
+  }
+  const setOnline = () => {
+    toggleOfflineModal(false)
+  }
+
+  useEffect(() => {
+    window.addEventListener("offline", setOffline)
+    window.addEventListener("online", setOnline)
+
+    return () => {
+      window.removeEventListener("offline", setOffline)
+      window.removeEventListener("online", setOnline)
+    }
+  }, [])
```

Comment:
If something happens and their network triggers on and off a few times this could be _really_ annoying. Maybe we should denounce it?

Reply:
Actually, thinking more about this, is there a subtler mechanism we could provide? 

There is a somewhat common pattern of using a banner to show offline

<img width="681" alt="image" src="https://user-images.githubusercontent.com/3087225/76093002-4c667800-5f8e-11ea-8ec1-d09536740034.png">

Reply:
Oh yeah! we can use the staging status-bar and change the color a bit. Good idea.